### PR TITLE
Updating NetOnly OneBranch

### LINKED ISSUE #4238: [Engineering Task] Create a Net Only release pipeline in OneBranch
This will be used to release the -net only version to the internal IDDP

### PR DESCRIPTION
**Pull request by trwalke**
Fixes #4238
Updating NetOnly OneBranch yaml to resolve oneBranch issues

Corresponding PR: https://identitydivision.visualstudio.com/IDDP/_git/MSAL.NET-OneBranch/pullrequest/9422

**Changes proposed in this request**

**Testing**

**Performance impact**

**Documentation**
- [ ] All relevant documentation is updated.

### CODE REVIEW

File: build/template-OneBranch-releasebuild-netonly.yaml [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-variables:
-  BuildPlatform: 'AnyCPU'
-  BuildConfiguration: 'Release'
-  ProjectFilePath: '$(Build.SourcesDirectory)\src\client\Microsoft.Identity.Client\Microsoft.Identity.Client.csproj'
-  ProjectFolderPath: '$(Build.SourcesDirectory)\src\client\Microsoft.Identity.Client'
+parameters:
+  BuildPlatform: 'any cpu'
+  BuildConfiguration: 'release'
+  MsalClientSemVer: '4.55.0-devopsbuild'
+  MsalSourceDir: 'microsoft-authentication-library-for-dotnet\' #Default MSAL Location
```
Comment (bgavrilMS) [inline build/template-OneBranch-releasebuild-netonly.yaml:7]: doesn't seem used? I see `ProjectFolderPath` still being referenced?

File: build/template-OneBranch-releasebuild-netonly.yaml [hunk 2]
```diff
@@ -1,41 +1,41 @@
-
 # Release build for -netonly package should be triggered after the normal release build
 parameters:
-  BuildPlatform: 'any cpu'
-  BuildConfiguration: 'release'
+  BuildPlatform: 'AnyCPU'
+  BuildConfiguration: 'Release'
   MsalClientSemVer: '4.55.0-devopsbuild'
-  MsalSourceDir: 'microsoft-authentication-library-for-dotnet\' #Default MSAL Location
+  ProjectFilePath: 'microsoft-authentication-library-for-dotnet\src\client\Microsoft.Identity.Client\Microsoft.Identity.Client.csproj'
```
Comment (bgavrilMS) [inline build/template-OneBranch-releasebuild-netonly.yaml:6]: Do these need to be input params? Would we ever change them?

**Review by bgavrilMS — APPROVED** [review]

**Review by gladjohn — APPROVED** [review]
please check if the [snupkg](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/f3a07ea0bf2cdc3178f373152f65f184c599f015/build/template-OneBranch-releasebuild-netonly.yaml#L79) is also signed

File: build/template-OneBranch-releasebuild-netonly.yaml [hunk 3]
```diff
@@ -1,42 +1,41 @@
-
 # Release build for -netonly package should be triggered after the normal release build
-variables:
+parameters:
   BuildPlatform: 'AnyCPU'
   BuildConfiguration: 'Release'
-  ProjectFilePath: '$(Build.SourcesDirectory)\src\client\Microsoft.Identity.Client\Microsoft.Identity.Client.csproj'
-  ProjectFolderPath: '$(Build.SourcesDirectory)\src\client\Microsoft.Identity.Client'
+  MsalClientSemVer: '4.55.0-devopsbuild'
+  ProjectFilePath: 'microsoft-authentication-library-for-dotnet\src\client\Microsoft.Identity.Client\Microsoft.Identity.Client.csproj'
```
Comment (gladjohn) [inline build/template-OneBranch-releasebuild-netonly.yaml:6]: You can just make this `ProjectFile` and use the `ProjectFolderPath` to get the full path

**Review by gladjohn — APPROVED** [review]

**Pull request merged**